### ISSUE TITLE
Diffusion fails when kld_weight is raised

### ISSUE THREAD

**Issue by Bathsheba** [princeton-computational-imaging/Diffusion-SDF#25]
I built a dataset and trained all 3 stages on it. It gives clean reconstructions but no new items or interpolations: generation after stage 3 is always either a data item, or hash.  I want to improve interpolation, but when I raise _kld_weight_ , stage 1 and stage 2 train well and produce accurate reconstruction, but stage 3 fails.

When β is raised, the encoder produces more widely varying latents for the same data item. (For β 1e-5 L1 between latents is ~.05, for β 1e-4 it's ~.25, the trend continues for higher β.) When diffusion is trained on 1 set of extracted latents, it reconstructs them well and it can generate data items from noise. 

But in end-to-end training, when the diffusion stage sees newly generated latents it cannot reconstruct them at all. It performs worse than when generating from noise, producing only hash. It never learns, so gensdf loss never stabilize or drops.

Any idea what this problem is? 
Thanks for your consideration.

**Comment by gene-chou** [comment]
We also had to spend considerable amount of time finetuning the loss functions and weights for our dataset. Honestly you might just need more data -- that always helps. Our method does not seem to interpolate well on small datasets, or may require a grid search for hyperparameters.

**Comment by Bathsheba** [comment]
Fair enough, thanks for your candor! 

I can't easily increase my dataset size...I wonder whether training diffusion on multiple sets of latents could help. I hate to abandon this encoder, it's the most economical and effective that I have seen, for compressing my topologically complex data.
